src/goto-programs/goto_convert_class.h

```diff
@@ -47,7 +48,10 @@ class goto_convertt:public messaget
   namespacet ns;
   std::string tmp_symbol_prefix;
 
-  void goto_convert_rec(const codet &code, goto_programt &dest);
+  void goto_convert_rec(
+    const codet &code,
+    goto_programt &dest,
+    const irep_idt &mode);
```

Comment:
I wonder if easier to set up a `goto_convertert` with a mode rather than adding an extra param everywhere

Reply:
Having it as a leaves the possibility to use the same converter for functions of different languages, and could make it easier to make some functions static. Now that the changes are done, I would say that it is easier to take that solution.